core/app/models/spree/image.rb

```diff
@@ -19,9 +19,18 @@ class Image < Asset
 
     # used by admin products autocomplete
     def mini_url
+      Spree::Deprecation.warn('Spree::Image#mini_url is DEPRECATED.')
       attachment.url(:mini, false)
     end
 
+    def filename
+      attachment_file_name
+    end
+
+    def url(size)
+      attachment.url(size)
```

Comment:
We are not able to decouple from paperclip if we have this method inside of here. It would be better to put this url retrieving logic inside the gallery classes. Extensions like `solidus_paperclip` would then implement such gallery classes.